### PR TITLE
Support iOS screenshot 1/2

### PR DESCRIPTION
**Pull request by xster**
#5150

- Use idevicescreenshot
- Add tests
- Remove global which path cache. Static fields are untestable and is probably not a good feature anyway if paths change

### CODE REVIEW

**Comment by xster** [comment]
We get #7800 for free

File: packages/flutter_tools/test/src/ios/devices_test.dart [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+    setUp(() {
+      mockProcessManager = new MockProcessManager();
+      mockOutputFile = new MockFile();
+    });
+
+    testUsingContext(
+      'screenshot without ideviceinstaller error',
+      () async {
+        when(mockOutputFile.path).thenReturn('/some/test/path/image.png');
```
Comment (cbracken) [inline packages/flutter_tools/test/src/ios/devices_test.dart]: nit: here and below: even though I suspect we'll never support iOS screenshotting on Windows, and this is a mock either way probably a good idea to to use `path.join()` to keep the codebase as cross-platform as possible.

File: packages/flutter_tools/test/src/ios/devices_test.dart [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        verify(mockProcessManager.runSync(
+          <String>['which', 'idevicescreenshot'], environment: null, workingDirectory: null));
+        verifyNever(mockProcessManager.runSync(
+          <String>['idevicescreenshot', '/some/test/path/image.png'],
+          environment: null,
+          workingDirectory: null
+        ));
+        expect(testLogger.errorText, contains('brew install ideviceinstaller'));
+      },
+      overrides: <Type, Generator>{ ProcessManager: () => mockProcessManager }
```
Comment (cbracken) [inline packages/flutter_tools/test/src/ios/devices_test.dart]: Here and below no spaces before/after `{}` in map literals.

**Review by cbracken — APPROVED** [review]

**Comment by cbracken** [comment]
lgtm

File: packages/flutter_tools/test/src/ios/devices_test.dart (hunk 1 above)
Comment (xster) [inline packages/flutter_tools/test/src/ios/devices_test.dart]: Ah sg

File: packages/flutter_tools/test/src/ios/devices_test.dart (hunk 2 above)
Comment (xster) [inline packages/flutter_tools/test/src/ios/devices_test.dart]: Done

**Pull request merged**